### PR TITLE
Process datetime on graphs

### PR DESCRIPTION
**Pull request by neuromantic99**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@superflows/chat-ui-react",
-  "version": "1.1.3",
+  "version": "1.1.4",
```
Comment (Henry-Pulver) [inline package.json:3]: I am sehr confused because this is already 1.1.4 according to main - think this needs a rebase! 😄

**Review by Henry-Pulver — COMMENTED** [review]
Realise this isn't explicitly a problem with the changes you've just made, but think we could possibly do something smarter when working out the x-axis of the graph. For Chattermill, the data is an array of objects with a `score` and an `updated_at`, but sadly it graphs with index on the x-axis
Screenshot from 2023-08-22 14-09-13
Screenshot from 2023-08-22 14-09-07

**Comment by neuromantic99** [comment]
> Realise this isn't explicitly a problem with the changes you've just made, but think we could possibly do something smarter when working out the x-axis of the graph. For Chattermill, the data is an array of objects with a `score` and an `updated_at`, but sadly it graphs with index on the x-axis Screenshot from 2023-08-22 14-09-13 Screenshot from 2023-08-22 14-09-07

we just need to add updated_at to possibleXlabels no? Or am i missing something more complex?

File: src/components/graph.tsx [hunk 2]
```diff
@@ -8,20 +8,111 @@ import {
...
 export function Graph(props: GraphData) {
   // Title currently does nothing in recharts, though there's an open
   // ticket for it
+
+  const xRange =
+    Math.max(...props.data.map((obj) => obj.x as number)) -
+    Math.min(...props.data.map((obj) => obj.x as number));
+
+  // This doesn't seem to be working for some reason
+  const offset = Math.ceil(xRange * 0.1);
```
[the remark this answers is not in the document]
Comment (neuromantic99) [reply src/components/graph.tsx:93]: no, it works on the usage page to offset the start and end of the x axis so it doesn't start exactly on the first and last datapoint but i dunno why it doesn't work here, may work for some datatypes and not others so i left it in

**Comment by Henry-Pulver** [comment]
> > Realise this isn't explicitly a problem with the changes you've just made, but think we could possibly do something smarter when working out the x-axis of the graph. For Chattermill, the data is an array of objects with a `score` and an `updated_at`, but sadly it graphs with index on the x-axis Screenshot from 2023-08-22 14-09-13 Screenshot from 2023-08-22 14-09-07
> 
> we just need to add updated_at to possibleXlabels no? Or am i missing something more complex?

I would first check for the possibleXlabels and then maybe check for if any columns correspond to times? Can also add `updated_at` to the list too :shrug:

File: src/components/graph.tsx [hunk 3]
```diff
@@ -125,8 +226,13 @@ export function extractGraphData(data: string): GraphData | null {
   // TODO: Currently don't support arrays of arrays
   if (typeof array[0] === "object" && !(array[0] instanceof Array)) {
     // Fields that match the possible x labels and are in every object in the array
+    // x matches are either in the possible x labels or can be converted to a date
```
Comment (Henry-Pulver) [inline src/components/graph.tsx:229]: Repetition?

File: src/components/graph.tsx [hunk 4]
```diff
@@ -125,8 +226,13 @@ export function extractGraphData(data: string): GraphData | null {
...
     const xMatches = Object.keys(array[0])
-      .filter((key) => checkStringMatch(key, possibleXlabels))
+      .filter(
+        (key) =>
+          checkStringMatch(key, possibleXlabels) ||
+          attemptDatetimeConversion(array.map((obj) => obj[key])) !== null
+      )
```
Comment (Henry-Pulver) [inline src/components/graph.tsx:235]: I don't know why, but this doesn't fix the Chattermill example :thinking: 

This is an example of the format: `2023-08-02T14:02:16.666Z`

Can send you what you need to repro if you like?

Screenshot from 2023-08-23 10-16-03

**Review by Henry-Pulver — COMMENTED** [review]
The x-axis change is great! :tada:

File: package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@superflows/chat-ui-react",
-  "version": "1.1.5",
+  "version": "1.1.6-8",
```
Comment (Henry-Pulver) [inline package.json:3]: :eyes:

**Pull request merged**